### PR TITLE
configure separate test output directory

### PR DESCRIPTION
**Pull request by brychcy**
for use with https://bugs.eclipse.org/bugs/show_bug.cgi?id=531072 (which isn't merged yet), so the MANIFEST.MF should probably be updated

### CODE REVIEW

File: org.jboss.tools.maven.apt.core/src/org/jboss/tools/maven/apt/internal/AbstractAptConfiguratorDelegate.java [hunk 1]
```diff
@@ -185,7 +196,7 @@ public void configureProject(IProgressMonitor monitor) throws CoreException {
       }
     }
 
-    Map<String, String> currentOptions = AptConfig.getProcessorOptions(javaProject);
+    Map<String, String> currentOptions = AptConfig.getRawProcessorOptions(javaProject);
```
Comment (fbricon) [inline org.jboss.tools.maven.apt.core/src/org/jboss/tools/maven/apt/internal/AbstractAptConfiguratorDelegate.java]: why use getRawProcessorOptions? it seems more limiting than getProcessorOptions

**Comment by fbricon** [comment]
Releasing a new version of m2e-apt uninstallable for Eclipse < 4.8 is gonna cause more trouble than necessary.
We'll probably want to enable the new generated test sources folder only if the new apt plugins is available, and keep the rest backward compatible.

File: org.jboss.tools.maven.apt.core/src/org/jboss/tools/maven/apt/internal/AbstractAptConfiguratorDelegate.java (hunk 1 above)
Comment (brychcy) [reply org.jboss.tools.maven.apt.core/src/org/jboss/tools/maven/apt/internal/AbstractAptConfiguratorDelegate.java]: getProcessorOptions includes computed options  -classpath, -sourcepath, -s, -d, so the following if() will alway fail. 
also, the computed options are now different for main and test compilation

**Comment by brychcy** [comment]
ok, i've updated to patch so it works with old eclipse by using reflection

File: org.jboss.tools.maven.apt.core/src/org/jboss/tools/maven/apt/internal/AbstractAptConfiguratorDelegate.java [hunk 2]
```diff
@@ -147,11 +159,24 @@ public void configureProject(IProgressMonitor monitor) throws CoreException {
...
-
-    AptConfig.setGenSrcDir(javaProject, generatedSourcesRelativeDirectoryPath);
+    if(generatedSourcesDirectory != null) {
+      // Configure APT output path
+      File generatedSourcesRelativeDirectory = convertToProjectRelativePath(eclipseProject, generatedSourcesDirectory);
+      String generatedSourcesRelativeDirectoryPath = generatedSourcesRelativeDirectory.getPath();
+  
+      AptConfig.setGenSrcDir(javaProject, generatedSourcesRelativeDirectoryPath);
+    }
+    if(generatedTestSourcesDirectory != null) {
```
Comment (fbricon) [inline org.jboss.tools.maven.apt.core/src/org/jboss/tools/maven/apt/internal/AbstractAptConfiguratorDelegate.java]: if(generatedTestSourcesDirectory != null && setGenTestSrcDirMethod != null)

**Comment by brychcy** [comment]
OK, merged the null check into the containing if

**Comment by fbricon** [comment]
Works as advertised with the latest jdt build and doesn't break in Oxygen, nice work @brychcy!

**Pull request merged**